src/web/demo.rkt

```diff
@@ -328,19 +328,58 @@
        (when (eof-object? formula)
          (raise-herbie-error "no formula specified"))
        (parse-test formula)
-       (define hash (sha1 (open-input-string formula-str)))
+       (define hash (sha1 (open-input-string (~s 'improve formula (get-seed)))))
        (body hash formula))]
     [_
      (response/error "Demo Error"
                      `(p "You didn't specify a formula (or you specified several). "
                          "Please " (a ([href ,go-back]) "go back") " and try again."))]))
 
+(define (improve-start-extract post-data)
+  (define form-str (hash-ref post-data 'formula))
+  (define form-seed (hash-ref post-data 'seed))
+  (list form-str form-seed))
+
+(define (improve-start-helper req body go-back)
+  (define post-body (request-post-data/raw req))
+  (define post-data (cond (post-body (bytes->jsexpr post-body)) (#t #f)))
```

Comment:
```suggestion
  (define post-data (and post-body (bytes->jsexpr post-body)))
```

Reply:
Was getting errors with the `and` version you proposed. 
I'm guessing that and was a way of checking that `post-body` was something and not `#f` as a sort of null check?